core/src/main/java/hudson/util/SubClassGenerator.java

```diff
@@ -1,99 +0,0 @@
-/*
- * The MIT License
- *
- * Copyright (c) 2010, InfraDNA, Inc.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-package hudson.util;
-
-import hudson.PluginManager.UberClassLoader;
-import jenkins.model.Jenkins;
-import org.kohsuke.asm5.ClassWriter;
-import org.kohsuke.asm5.MethodVisitor;
-import org.kohsuke.asm5.Type;
-
-import java.lang.reflect.Constructor;
-
-import static org.kohsuke.asm5.Opcodes.ACC_PUBLIC;
-import static org.kohsuke.asm5.Opcodes.ALOAD;
-import static org.kohsuke.asm5.Opcodes.ILOAD;
-import static org.kohsuke.asm5.Opcodes.INVOKESPECIAL;
-import static org.kohsuke.asm5.Opcodes.RETURN;
-
-/**
- * Generates a new class that just defines constructors into the super types.
- *
- * @author Kohsuke Kawaguchi
- */
-public class SubClassGenerator extends ClassLoader {
```

Comment:
By the way did you check the usage-in-plugins tool? I have not found any usages in @jenkinsci nor @cloudbees according to GitHub text search, though that is not totally reliable. Some interesting things I stumbled across:

* https://github.com/yannrouillard/pkg-jenkins/blob/f63a94ed21bd1584ab3043c33a4b22ee05a31b7e/debian/patches/build/0015-use-stock-asm.patch
* https://github.com/jenkins-infra/unused-code-detector/blob/b14b5cbf2880178cc7b48cecb846b66530ea7c03/Output_example.html#L1223

Reply:
Yup, I ran `usage-in-plugins` against `hudson/util/SubClassGenerator` for update centers https://jenkins-updates.cloudbees.com/update-center/envelope-core-oc/update-center.json?version=2.277.4.3 and https://jenkins-updates.cloudbees.com/update-center/envelope-core-mm/update-center.json?version=2.277.4.3; found no usages.

Reply:
but what about actual shaded ASM classes?